### DISCUSSION TITLE
Plot 2d with pyvista

### DISCUSSION THREAD

**Discussion by HumbertHumbert7** [pyvista/pyvista#5992] [Q&A] [answered]
Hi, is it possible to realize a classical 2d contour plot with pyvista? If it is possible how can I do? (I can add my code if needed).

**Comment by tkoyama010** [comment]
Please refer code in this link.
https://banesullivan.com/pyvista/examples/contouring.html
![image](https://banesullivan.com/_images/sphx_glr_contouring_002.png)

**Comment by HumbertHumbert7** [comment]
Hi tkoyama010, thank you for your reply.
![plasticityplot](https://github.com/pyvista/pyvista/assets/161707314/030260ab-3643-4064-b22b-960ba3ef1ccd)
That's my plot obtained with pyvista (I can also add the code, if you need to see it). I just would like, if possible, to mantain it in this way and just add the axis and possibly a grid. Do you know how can I do it?

**Reply by tkoyama010** [reply]
You can find instructions on how to add contour lines to a mesh here.
https://tutorial.pyvista.org/tutorial/04_filters/solutions/d_contouring.html

**Reply by HumbertHumbert7** [reply]
Thank you tkoyama010, however I would like to know if there is the possibility to obtain a plot like this with pyvista
![IMG_6330](https://github.com/pyvista/pyvista/assets/161707314/185407ba-7303-4d4d-be4b-d7f171af1109)
So, just add the axes in this way.
(This one was done with matplotlib).

**Reply by tkoyama010** [reply, marked as answer]
You can show the bounds axis using the `show_bounds` method. See [`show_bounds` method API document](https://docs.pyvista.org/version/stable/api/plotting/_autosummary/pyvista.Plotter.show_bounds.html) for more detail.
(Edit: To make a 2D view, you can control `cpos` arg in the show method. See [`show` method API document](https://docs.pyvista.org/version/stable/api/plotting/_autosummary/pyvista.plotter.show) for more detail. Also, [`enable_parallel_projection`](https://docs.pyvista.org/version/stable/api/plotting/_autosummary/pyvista.Plotter.enable_parallel_projection.html) is also useful to make it a 2D view.)
```python
import pyvista as pv
from pyvista import examples

mesh = pv.Sphere()
plotter = pv.Plotter()
actor = plotter.add_mesh(mesh)
actor = plotter.show_bounds(
    grid='front',
    location='outer',
    all_edges=True,
)
plotter.enable_parallel_projection()
plotter.show(cpos="xy")
```
![image](https://github.com/pyvista/pyvista/assets/7513610/cf785139-c322-45f5-a238-58e549b5f04e)
